presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -225,7 +229,25 @@ private Table getTable(SemiTransactionalHiveMetastore metastore, SchemaTableName
                     filter.add(HivePartitionKey.HIVE_DEFAULT_DYNAMIC_PARTITION);
                 }
                 else if (value instanceof Slice) {
-                    filter.add(((Slice) value).toStringUtf8());
+                    Type type = domain.getType();
+                    Slice slice = (Slice) value;
+                    if (type instanceof CharType) {
+                        int length = ((CharType) type).getLength();
+                        StringBuilder builder = new StringBuilder(length);
+                        String sliceString = slice.toStringUtf8();
+                        builder.append(sliceString);
+                        for (int i = sliceString.length(); i < length; i++) {
```

Comment:
extract to a method (possibility in some reusable place)... unless there isn't one already?